apps/admin_app/lib/admin_app_web/controllers/order_controller.ex

```diff
@@ -1,26 +1,211 @@
 defmodule AdminAppWeb.OrderController do
   use AdminAppWeb, :controller
   alias Snitch.Data.Model.Order
+  alias BeepBop.Context
+  alias Snitch.Domain.Order.DefaultMachine
+  alias Snitch.Data.Schema.Variant
+  alias Snitch.Data.Schema.Address
   alias Snitch.Repo
+  import Ecto.Query
 
   def index(conn, _params) do
     render(conn, "index.html", %{orders: Repo.preload(Order.get_all(), :user)})
   end
 
-  def show(conn, %{"slug" => slug} = _params) do
-    order =
-      %{slug: slug}
-      |> Order.get()
-      |> Repo.preload(line_items: [:variant], billing_address: [], shipping_address: [])
+  def show(conn, params) do
+    order = load_order(%{number: params["number"]})
 
-    render(conn, "show.html", %{order: order})
+    search =
+      case params["search"] do
+        nil ->
+          []
+
+        _ ->
+          search_item_variant(params["search"])
+      end
+
+    render(conn, "show.html", %{order: order, search: search})
+  end
+
+  def create(conn, _params) do
+    current_user = Guardian.Plug.current_resource(conn)
+    {:ok, order} = Order.create(%{line_items: [], user_id: current_user.id})
+
+    redirect(conn, to: "/orders/#{order.number}")
```

Comment:
why `"/orders/#{order.number}"` is required. why not like this `redirect(to: order_path(conn, :show, order.id))`